### PR TITLE
Update Aspire to 9.0.0-rc.1

### LINKED ISSUE #403: Decide strategy for .NET 9 and Aspire 9 in samples
.NET 9 and Aspire 9 will be released in November this year. Aspire 9 will support both .NET 8 and .NET 9, i.e. the most recent in-support LTS and STS releases of .NET. Once Aspire 9 is released it will be the only supported version of Aspire as it is a "tip" product with only the latest stable version being supported

We need to decide what to do for the samples in this repo when this happens. They should all be moved to use Aspire 9 as part of the release as that will be the most recent version of Aspire and thus the only supported version. But the question of whether to have the samples continue to target .NET 8, move to .NET 9, or multi-target or be replicated for both, is an open one.

Cc @maddymontaquila

### PR DESCRIPTION
**Pull request by DamianEdwards**
- Updates samples to the public Aspire 9.0.0-rc.1 release and uses features like `WaitFor` where appropriate.
- Bumps all dependencies to latest version.
- Added a missing README file.
- Other minor tweaks

Fixes #403

### CODE REVIEW

File: samples/AspireWithJavaScript/AspireJavaScript.AppHost/aspire-manifest.json [hunk 1]
```diff
@@ -47,9 +51,10 @@
       "path": "../AspireJavaScript.React/Dockerfile",
       "context": "../AspireJavaScript.React",
       "env": {
-        "NODE_ENV": "production",
-        "REACT_APP_WEATHER_API_HTTP": "{weatherapi.bindings.http.url}",
-        "REACT_APP_WEATHER_API_HTTPS": "{weatherapi.bindings.https.url}",
+        "NODE_ENV": "development",
```
Comment (DamianEdwards) [inline samples/AspireWithJavaScript/AspireJavaScript.AppHost/aspire-manifest.json:54]: This file just hadn't been updated for quite a while. The change to the environment variables used here were made many months ago separately to this PR.

File: samples/AspireWithPython/InstrumentedPythonProject/main.py [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 
+trace.set_tracer_provider(TracerProvider())
+otlpExporter = OTLPSpanExporter()
+processor = BatchSpanProcessor(otlpExporter)
+trace.get_tracer_provider().add_span_processor(processor)
+
+FlaskInstrumentor().instrument_app(app)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
```
Comment (DamianEdwards) [inline samples/AspireWithPython/InstrumentedPythonProject/main.py:20]: This change adds OTel tracing support.

File: samples/ContainerBuild/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+name: ".NET Aspire Container Build Sample"
+urlFragment: "aspire-container-build"
+description: "An example of integrating apps written in other languages using container-based builds in a .NET Aspire app."
+---
+
+# Working with container-built resources in a .NET Aspire application
+
+This sample demonstrates integrating applications into a .NET Aspire app via Dockerfiles and container-based builds. This is especially helpful to integrate applications written in languages that .NET Aspire does not have a native integration for, or to reduce the prerequisites required to run the application.
+
+![Screenshot of the Aspire dashboard showing the ginapp container resource built from a Dockerfil](./images/aspire-dashboard-container-build.png)
```
Comment (DamianEdwards) [inline samples/ContainerBuild/README.md:18]:
```suggestion
![Screenshot of the Aspire dashboard showing the ginapp container resource built from a Dockerfile](./images/aspire-dashboard-container-build.png)
```

File: samples/ContainerBuild/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+The sample integrates a simple app written using [Go](https://go.dev/) and the [Gin Web Framework](https://gin-gonic.com/) by using a [Dockerfile](./ginapp/Dockerfile):
+
+- **ginapp**: This is a simple "Hello, World" HTTP API that returns a JSON object like `{ "message": "Hello, World!" }` from `/`.
+
+## Pre-requisites
+
+- [.NET 8 SDK](https://dotnet.microsoft.com/download/dotnet/8.0)
+- [An container runtime supported by .NET Aspire, e.g. Docker Desktop or Podman](https://learn.microsoft.com/dotnet/aspire/fundamentals/setup-tooling#container-runtime)
```
Comment (DamianEdwards) [inline samples/ContainerBuild/README.md:27]:
```suggestion
- [A container runtime supported by .NET Aspire, e.g. Docker Desktop or Podman](https://learn.microsoft.com/dotnet/aspire/fundamentals/setup-tooling#container-runtime)
```

File: samples/DatabaseMigrations/DatabaseMigrations.MigrationService/Program.cs [hunk 5]
```diff
@@ -9,9 +9,6 @@
 
 builder.AddServiceDefaults();
 
-builder.Services.AddOpenTelemetry()
```
Comment (DamianEdwards) [inline samples/DatabaseMigrations/DatabaseMigrations.MigrationService/Program.cs:12]: Not needed anymore as the Service Defaults is adding a source for the application name which the `DbInitializer` uses now.

File: tests/SamplesIntegrationTests/AppHostTests.cs [hunk 6]
```diff
@@ -45,8 +44,7 @@ public async Task TestEndpointsReturnOk(TestEndpoints testEndpoints)
         var projects = appHost.Resources.OfType<ProjectResource>();
         await using var app = await appHost.BuildAsync();
 
-        await app.StartAsync();
-        await app.WaitForResources().WaitAsync(TimeSpan.FromSeconds(30));
+        await Task.WhenAll(app.StartAsync(), app.WaitForResources()).WaitAsync(TimeSpan.FromSeconds(120));
```
Comment (DamianEdwards) [inline tests/SamplesIntegrationTests/AppHostTests.cs:47]: This was a key change as `app.StartAsync()` now includes waiting for all resources to get to the starting state which includes any `WaitFor` registrations.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**